javacutil/src/org/checkerframework/javacutil/TypesUtils.java

```diff
@@ -132,9 +133,24 @@ public static boolean isThrowable(TypeMirror type) {
     }
 
     /**
+     * Returns true iff the argument is an anonymous type.
+     *
+     * @return whether the argument is an anonymous type
+     */
+    public static boolean isAnonymous(TypeMirror type) {
+        if (type instanceof DeclaredType) {
```

Comment:
I would rewrite that as one if with && of the two conditions or, even simpler, a `return A && B`;